### PR TITLE
Apply guidelines, improve tests and docs, support solo staker migration

### PR DESCRIPTION
**Pull request by DrZoltanFazekas**
Add more tests and assertions to the existing tests, particularly in the non-liquid staking variant. Also extend the readme file, describe which methods and events can be used by dapps integrating with the delegation contracts. Furthermore, ensure compatibility with `deposit_v3.sol`.

### CODE REVIEW

File: src/BaseDelegation.sol [hunk 1]
```diff
@@ -201,6 +183,20 @@ abstract contract BaseDelegation is Delegation, PausableUpgradeable, Ownable2Ste
...
+    function getPendingClaims() public virtual view returns(uint256[2][] memory claims) {
+        BaseDelegationStorage storage $ = _getBaseDelegationStorage();
+        WithdrawalQueue.Fifo storage fifo = $.withdrawals[_msgSender()];
+        uint256 index = fifo.first;
+        while (fifo.ready(index))
+            index++;
+        uint256 firstPending = index;
+        claims = new uint256[2][](fifo.last - index);
+        while (fifo.notReady(index)) {
+            claims[index - firstPending] = [fifo.items[index].blockNumber, fifo.items[index].amount];
```
Comment (86667) [inline src/BaseDelegation.sol:195]: nit: paying gas cost of accessing `fifo.items` from storage twice

File: state.sh [hunk 2]
```diff
@@ -16,75 +18,76 @@ if [ $# -eq 3 ]; then
...
         price=$(bc -l <<< "scale=36; ($stake+$rewardsBeforeUnstaking-($rewardsBeforeUnstaking-$taxedRewardsBeforeUnstaking)*$commissionNumerator/$denominator)/$totalSupply")
     else
         price=$(bc -l <<< "scale=36; 1/1")
     fi
-    price0=$(cast call $1 "getPrice()(uint256)" --block $block_num --rpc-url http://localhost:4201 | sed 's/\[[^]]*\]//g')
-    echo LST supply: $(cast to-unit $totalSupply ether) ZIL
-    echo LST price: $price \~ $(cast to-unit $price0 ether)
-    echo staker LST value: $(bc -l <<< "scale=18; $staker_lst*$price") ZIL
+    price0=$(cast call $1 "getPrice()(uint256)" --block $block_num --rpc-url $url | sed 's/\[[^]]*\]//g')
+    echo $symbol supply: $(cast to-unit $totalSupply ether)
```
Comment (86667) [inline state.sh:69]: ZIL removed from the end on purpose?

Comment (DrZoltanFazekas) [reply state.sh:69]: yes, the LST symbol can be anything but not ZIL, it was incorrect before

File: README.md [hunk 3]
```diff
@@ -86,28 +86,48 @@ cast call 0x7A0b7e6D24eDe78260c9ddBD98e828B0e11A8EA2 "DENOMINATOR()(uint256)" --
...
-0x7a0b7e6d24ede78260c9ddbd98e828b0e11a8ea2 "deposit(bytes,bytes,bytes)" \
+0x7a0b7e6d24ede78260c9ddbd98e828b0e11a8ea2 "depositFirst(bytes,bytes,bytes)" \
 0x92fbe50544dce63cfdcc88301d7412f0edea024c91ae5d6a04c7cd3819edfc1b9d75d9121080af12e00f054d221f876c \
 0x002408011220d5ed74b09dcbe84d3b32a56c01ab721cf82809848b6604535212a219d35c412f \
 0xb14832a866a49ddf8a3104f8ee379d29c136f29aeb8fccec9d7fb17180b99e8ed29bee2ada5ce390cb704bc6fd7f5ce814f914498376c4b8bc14841a57ae22279769ec8614e2673ba7f36edc5a4bf5733aa9d70af626279ee2b2cde939b4bd8a
 ```
-with the BLS public key, the peer id and the BLS signature of your node. Note that the peer id must be converted from base58 to hex. Make sure your node is fully synced before you run the above command.
+with the BLS public key, the peer id and the BLS signature of your node. Note that the peer id must be converted from base58 to hexadecimal format and you must provide the delegation contract address when generating the BLS signature:
+```bash
+echo '{"secret_key":"...", "chain_id":..., "address":"0x7a0b7e6d24ede78260c9ddbd98e828b0e11a8ea2"}' | cargo run --bin convert-key
```
[the remark this answers is not in the document]
Comment (DrZoltanFazekas) [reply README.md]: oh, you're right, we changed it today

**Review by 86667 — APPROVED** [review]

**Pull request merged**